### DISCUSSION TITLE
Sync with delete - should it clear empty dirs?

### DISCUSSION THREAD

**Discussion by themagic314** [umputun/spot#66] [General]
After using the ```sync``` task with ```"delete": true```, empty nested folders are not removed when the last file is deleted. 
Users can solve this by using an additional task with a command like ```find /dir/ -empty -type d -delete``` after sync.

The question is whether this is intended behavior or ```sync``` should perform a "full" sync, including deleting empty directories.

**Comment by umputun** [comment]
This is not really intentional however, from what I recall rsync won't delete such directories either unless `--prune-empty-dirs` is set. This is possible to implement, but I'm not sure if making it part of `"delete": true` is a good idea. For example, this may prevent creating empty nested folder by sync command

**Comment by umputun** [comment]
Latest 0.14.2 has the fix to remove the remote directory if no such directory on the local side (if delete is true)

**Reply by umputun** [reply]
note: another nice side-effect of this fix - the `delete` command in non-recursive mode refused to remove an empty directory. Now it will, but only if the directory is empty. To remove non-empty directories recursive flag should be used, as before
